docker/puppetdb/ssl-setup.sh

```diff
@@ -0,0 +1,420 @@
+#
+# Example:
+#
+#    appendline "mysetting = myvalue" /etc/myconfig
+#
+function appendline {
+  backupfile $2
+  tmp=$2.tmp.`date +%s`
+  cat $2 > ${tmp}
+  echo $1 >> ${tmp}
+  mv ${tmp} $2
+}
+
+# This function copies the necessary PEM files from Puppet to the PuppetDB
+# SSL directory.
+#
+# This expects various environment variables to have already been set to work.
+function copy_pem_from_puppet {
+  orig_files=($orig_ca_file $orig_private_file $orig_public_file)
+  for orig_file in "${orig_files[@]}"; do
+    if [ ! -e $orig_file ]; then
+      echo "Warning: Unable to find all puppet certificates to copy"
+      echo
+      echo "  This tool requires the following certificates to exist:"
+      echo
+      echo "  * $orig_ca_file"
+      echo "  * $orig_private_file"
+      echo "  * $orig_public_file"
+      echo
+      echo "  These files may be missing due to the fact that your host's Puppet"
+      echo "  certificates may not have been signed yet, probably due to the"
+      echo "  lack of a complete Puppet agent run. Try running puppet first, for"
+      echo "  example:"
+      echo
+      echo "      puppet agent --test"
+      echo
+      echo "  Afterwards re-run this tool then restart PuppetDB to complete the SSL"
+      echo "  setup:"
+      echo
+      echo "      ${ssl_command} -f"
+      exit 1
+    fi
+  done
+  rm -rf $ssl_dir
+  mkdir -p $ssl_dir
+  echo "Copying files: ${orig_ca_file}, ${orig_private_file} and ${orig_public_file} to ${ssl_dir}"
+  cp -pr $orig_ca_file $ca_file
+  cp -pr $orig_private_file $private_file
+  cp -pr $orig_public_file $public_file
+}
+
+########
+# MAIN #
+########
+
+# Gather command line options
+while getopts "ifh" opt;
+do
+  case $opt in
+    i)
+      interactive=true ;;
+    f)
+      force=true ;;
+    h)
+      usage ;;
+    *)
+      usage ;;
+  esac
+done
+
+${interactive:=false}
+${force:=false}
+
+# Deal with interactive setups differently to non-interactive
+if $interactive
+then
+  dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+  cd $dir
+  answers_file="puppetdb-ssl-setup-answers.txt"
+  if [ -f "$answers_file" ]
+  then
+    echo "Reading answers file '$answers_file'"
+    . $answers_file
+  fi
+
+  vars=( agent_confdir agent_vardir puppetdb_confdir )
+  prompts=( "Puppet Agent confdir" "Puppet Agent vardir" "PuppetDB confdir" )
+
+  for (( i=0; i<${#vars[@]}; i++ ))
+  do
+    read -p "${prompts[$i]} [${!vars[$i]}]: " input
+    export ${vars[$i]}=${input:-${!vars[$i]}}
+  done
+
+  cat /dev/null > $answers_file
+  for (( i=0; i<${#vars[@]}; i++ ))
+  do
+    echo "${vars[$i]}=${!vars[$i]}" >> $answers_file
+  done
+else
+  # This should be run on the host with PuppetDB
+  PATH=/opt/puppetlabs/bin:/opt/puppet/bin:$PATH
+  agent_confdir=`puppet agent --configprint confdir`
+  agent_vardir=`puppet agent --configprint vardir`
+  # user=<%= EZBake::Config[:user] %>
+  # group=<%= EZBake::Config[:group] %>
+  user=puppetdb
+  group=puppetdb
```

Comment:
This entire file is just the current `ssl-setup.erb` file, but with the 2 lines of EZBake replaced with hardcoded FOSS values.